components/script/script_runtime.rs

```diff
@@ -831,47 +832,67 @@ impl StreamConsumer {
     }
 }
 
+/// Implements the steps to compile webassembly response mentioned here
+/// <https://webassembly.github.io/spec/web-api/#compile-a-potential-webassembly-response
 #[allow(unsafe_code)]
-unsafe extern "C" fn ConsumeStreamCallback(
+unsafe extern "C" fn consume_stream(
     _cx: *mut RawJSContext,
     obj: HandleObject,
     _mimeType: MimeType,
     _consumer: *mut JSStreamConsumer,
-) {
+) -> bool {
     let cx = JSContext::from_ptr(_cx);
     let global = GlobalScope::from_context(*cx);
-    let returnValue = Promise::new(&global);
-    let mut handleobj = UndefinedValue();
-    GetProxyPrivate(*obj, &mut handleobj);
-    rooted!(in(*cx) let target = handleobj.to_object());
-    if let Ok(unwrappedSource) = root_from_handleobject::<Response>(target.handle(), *cx) {
-        let mimetype = unwrappedSource.Headers().extract_mime_type();
-        if std::str::from_utf8(&mimetype).unwrap() != "application/wasm" {
-            return returnValue.reject_error(Error::Type(
-                "Response has unsupported MIME type".to_string(),
-            ));
+
+    //Step 2.1 Upon fulfillment of source, store the Response with value unwrappedSource.
+    if let Ok(unwrapped_source) =
+        root_from_handleobject::<Response>(RustHandleObject::from_raw(obj), *cx)
+    {
+        //Step 2.2 Let mimeType be the result of extracting a MIME type from response’s header list.
+        let mimetype = unwrapped_source.Headers().extract_mime_type();
+
+        //Step 2.3 If mimeType is not `application/wasm`, return with a TypeError and abort these substeps.
+        if mimetype == b"application/wasm" {
+            throw_dom_exception(
+                cx,
+                &global,
+                Error::Type("Response has unsupported MIME type".to_string()),
+            );
```

Comment:
In each of the cases where we throw an exception, we need to return false instead of true.

Reply:
done